include/onnxruntime/core/graph/graph.h

```diff
@@ -780,7 +780,8 @@ class Graph {
 
   // Add node with specified <node_proto>.
   Node& AddNode(const ONNX_NAMESPACE::NodeProto& node_proto,
-                const ArgNameToTypeMap& name_to_type);
+                const ArgNameToTypeMap& name_to_type,
+                TypeToCountMap& type_to_count_map);
```

Comment:
>TypeToCountMap& type_to_count_map [](start = 16, length = 33)

Not a huge fan of the API taking this in order to create a hopefully unique name for the node as it's a bit obtuse. Possibly not expected behaviour for the name to be changed during AddNode either. 

Why do we need this new approach?

Reply:
Reverted the this aspect of the change